napari/_qt/qt_dims_slider.py

```diff
@@ -0,0 +1,471 @@
+        if value.is_integer():
+            value = int(value)
+        return str(value)
+
+
+class QtPlayButton(QPushButton):
+    """Play button, included in the DimSliderWidget, to control playback
+
+    the button also owns the QtModalPopup that controls the playback settings.
+    """
+
+    play_requested = Signal(int)  # axis, fps
+
+    def __init__(self, dims, axis, reverse=False, fps=10, mode=1):
+        super().__init__()
+        self.dims = dims
+        self.axis = axis
+        self.reverse = reverse
+        self.fps = fps
+        self.mode = mode
+        self.setProperty('reverse', str(reverse))  # for styling
+        self.setProperty('playing', 'False')  # for styling
+
+        # build popup modal form
+
+        self.popup = QtModalPopup(self)
+        fpsspin = QtCustomDoubleSpinBox(self.popup)
+        fpsspin.setAlignment(Qt.AlignCenter)
+        fpsspin.setValue(self.fps)
+        fpsspin.setStepType(QDoubleSpinBox.AdaptiveDecimalStepType)
+        fpsspin.setMaximum(500)
+        fpsspin.setMinimum(0)
+
+        revcheck = QCheckBox(self.popup)
+        layout = QHBoxLayout()
+        layout.addWidget(fpsspin)
+        layout.addWidget(revcheck)
+        self.popup.form_layout.insertRow(
+            0, QLabel('frames per sec:', parent=self.popup), layout
+        )
+        self.fpsspin = fpsspin
+        self.reverse_check = revcheck
+
+        # dimsrange = dims.dims.range[axis]
+        # minspin = QDoubleSpinBox(self.popup)
+        # minspin.setAlignment(Qt.AlignCenter)
+        # minspin.setValue(dimsrange[0])
+        # minspin.valueChanged.connect(self.set_minframe)
+        # self.popup.form_layout.insertRow(
+        #     1, QLabel('start frame:', parent=self.popup), minspin
+        # )
+
+        # maxspin = QDoubleSpinBox(self.popup)
+        # maxspin.setAlignment(Qt.AlignCenter)
+        # maxspin.setValue(dimsrange[1] * dimsrange[2])
+        # maxspin.valueChanged.connect(self.set_maxframe)
+        # self.popup.form_layout.insertRow(
+        #     2, QLabel('end frame:', parent=self.popup), maxspin
+        # )
+
+        self.mode_combo = QComboBox(self.popup)
+        self.mode_combo.addItems(['play once', 'loop', 'back and forth'])
+        self.popup.form_layout.insertRow(
+            1, QLabel('play mode:', parent=self.popup), self.mode_combo
+        )
+        self.mode_combo.setCurrentIndex(self.mode)
+
+    def mouseReleaseEvent(self, event):
+        # using this instead of self.customContextMenuRequested.connect and
+        # clicked.connect because the latter was not sending the
+        # rightMouseButton release event.
+        if event.button() == Qt.RightButton:
+            self.popup.show_above_mouse()
+        elif event.button() == Qt.LeftButton:
+            self._on_click()
+
+    def _on_click(self):
+        if self.property('playing') == "True":
+            return self.dims.stop()
+        # TODO: link this to DimSliderWidget not QtDims
```

Comment:
What is this TODO? Does it need to be done before merge? Can you add more detail in the comment

Reply:
changed my mind on this... works best as is, removing the TODO